tests/readiness_test.go

```diff
@@ -38,8 +38,42 @@ func TestReadiness(t *testing.T) {
 	clientset := runAgent(proxy.agent, stopCh)
 	waitForConnectedServerCount(t, 1, clientset)
 
+	// check the agent connected status
```

Comment:
I would rename this test `TestServerReadiness` and keep agent readiness tests separate.

Reply:
Or, maybe we should move this file to `tests/server_readiness.go` and introduce `tests/agent_readiness.go`

Reply:
I've renamed the tests to depict what they mean. Not sure if have two different files is a good idea.

for example: the tests would be test grpc server and agent readiness, test http connect server and agent readiness and test agent readiness without the server.

WDYT ?

Reply:
@ipochi did you upload your latest changes? I don't yet see the corresponding changes.

Reply:
@jkh52 Sorry about that. Pushed the changes.

I would like to keep the logic "Agent reports Ready, when it is connected to atleast one konnectivity-server".

If we have the logic of "Ready only when all servers are connected" then lets say one of the konnectivity-server is in a crashloop state, then even though tunnel is working fine in a degraded state, kubernetes marks the agent-pods as not ready.

Which may not be just the visual look as there might be other sub-systems that might kick in based on the non-readiness.

Reply:
@jkh52 Sorry about that. Pushed the changes.

I would like to keep the logic "Agent reports Ready, when it is connected to atleast one konnectivity-server".

If we have the logic of "Ready only when all servers are connected" then lets say one of the konnectivity-server is in a crashloop state, then even though tunnel is working fine in a degraded state, kubernetes marks the agent-pods as not ready.

Which may not be just the visual look as there might be other sub-systems that might kick in based on the non-readiness.

Reply:
+1 to simply changing the default meaning of readiness to "at least 1".

To repeat what @cheftako mentioned in this week's biweekly meeting: a user who may want the more complex definition "connected == ServerCount" can ask for it using a URL parameter (to be implemented later if desired). There is precedence elsewhere (kube-apiserver).